.github/actions/promote-to-stable/action.yml

```diff
@@ -34,22 +40,40 @@ runs:
     - name: Promote RPM packages to stable
       if: ${{ startsWith(inputs.distrib, 'el') }}
       run: |
-        set -x
+        set -eux
+
+        # DEBUG
         echo "[DEBUG] - Major version: ${{ inputs.major_version }}"
         echo "[DEBUG] - Minor version: ${{ inputs.minor_version }}"
         echo "[DEBUG] - Distrib: ${{ inputs.distrib }}"
+        echo "[DEBUG] - release_cloud: ${{ inputs.release_cloud }}"
+        echo "[DEBUG] - release_type: ${{ inputs.release_type }}"
 
         # Cloud specific promote
         # delivery by default to onprem, override to internal if base branch is master
         if [[ ${{ inputs.github_base_ref }} == "master" ]]; then
-          ROOT_REPO_PATH="rpm-standard-internal"
+          #ROOT_REPO_PATH="rpm-standard-internal"
+          ROOT_REPO_PATH="test-rpm-standard-internal"
         else
-          ROOT_REPO_PATH="rpm-standard"
+          #ROOT_REPO_PATH="rpm-standard"
+          ROOT_REPO_PATH="test-rpm-standard"
         fi
 
+        # Build search path based on release_cloud and release_type values
+        # if cloud, search in testing-<release_type> path
+        # if non-cloud, search in the testing usual path
+        if [[ ${{ inputs.release_cloud }} -eq 1 && ${{ inputs.release_type }} == "hotfix" ]] || [[ ${{ inputs.release_cloud }} -eq 1 && ${{ inputs.release_type }} == "release" ]];then
+          SEARCH_REPO_PATH="${{ inputs.major_version }}/${{ inputs.distrib }}/testing-${{ inputs.release_type }}/$ARCH/${{ inputs.module_name }}"
+        elif [[ ${{ inputs.release_cloud }} -eq 0 ]];then
+          SEARCH_REPO_PATH="${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module_name }}/"
```

Comment:
testing-release and testing-hotfix are not used for onprem ?

Reply:
will not be used for on prem
only cloud is involved in this